### PR TITLE
Option::orElse argument as Closure or mixed value support

### PR DESCRIPTION
**Pull request by Strate**
Implementation of #19

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,10 @@
         "phpunit/phpunit": "4.7.*"
     },
     "autoload": {
-        "psr-0": { "PhpOption\\": "src/" }
+        "psr-0": {
+            "PhpOption\\": "src/",
+            "PhpOption\\Tests\\": "tests/"
```
Comment (Strate) [inline composer.json]: This requuired for inheritance of test cases support

Comment (schmittjoh) [inline composer.json]: Please revert this, and instead add this to the autoloader in the bootstrap.php file for tests:

```php
$loader = require $autoloadFile;
$loader->add('PhpOption\\Tests\\', __DIR__);
```

File: src/PhpOption/LazyOption.php [hunk 2]
```diff
@@ -88,9 +88,9 @@ public function getOrThrow(\Exception $ex)
         return $this->option()->getOrThrow($ex);
     }
 
-    public function orElse(Option $else)
+    public function orElse($else, $noneValue = null)
```
Comment (schmittjoh) [inline src/PhpOption/LazyOption.php]: Let's not add noneValue here. If you want to use something else, but ``null`` you just have to pass an Option.

File: composer.json (hunk 1 above)
Comment (lstrojny) [inline composer.json]: Why not use https://getcomposer.org/doc/04-schema.md#autoload-dev ?

Comment (schmittjoh) [inline composer.json]: That's good point. Thanks :)

File: src/PhpOption/LazyOption.php (hunk 2 above)
Comment (Strate) [inline src/PhpOption/LazyOption.php]: Yes, or just a write something like:
```php
->orElse(...)
->reject($noneValue)
```

**Pull request closed without merging**